src/lib/AsyncTypeaheadInput.tsx

```diff
@@ -23,46 +24,62 @@ const AsyncTypeaheadInput = <T extends FieldValues>(props: AsyncTypeaheadProps<T
   const focusHandler = useMarkOnFocusHandler(markAllOnFocus);
 
   return (
-    <Controller
-      control={control}
-      name={name}
-      render={({ field, fieldState: { error } }) => (
-        <FormGroupLayout helpText={helpText} name={name} id={id} label={label} labelToolTip={labelToolTip}>
-          <AsyncTypeahead
-            {...field}
-            id={id}
-            multiple={props.multiple}
-            defaultSelected={defaultSelected}
-            onChange={(e) => {
-              const values = convertTypeaheadOptionsToStringArray(e);
-              const finalValue = props.multiple ? values : values[0];
+    <>
+      <Controller
+        control={control}
+        name={name}
+        render={({ field, fieldState: { error } }) => (
+          <FormGroupLayout helpText={helpText} name={name} id={id} label={label} labelToolTip={labelToolTip}>
+            <AsyncTypeahead
+              {...field}
+              id={id}
+              multiple={props.multiple}
+              defaultSelected={defaultSelected}
+              onChange={(e) => {
+                const values = convertTypeaheadOptionsToStringArray(e);
+                const finalValue = props.multiple ? values : values[0];
 
-              if (onChange) {
-                onChange(finalValue);
-              }
+                if (onChange) {
+                  onChange(finalValue);
+                }
 
-              field.onChange(finalValue);
-            }}
-            className={error ? "is-invalid" : ""}
-            inputProps={{ id }}
-            isLoading={isLoading}
-            options={options}
-            filterBy={() => true}
-            onSearch={(query) => {
-              void (async () => {
-                setIsLoading(true);
-                const results = await props.queryFn(query);
-                setOptions(results);
-                setIsLoading(false);
-              })();
-            }}
-            disabled={disabled}
-            onFocus={focusHandler}
-            {...reactBootstrapTypeaheadProps}
-          />
-        </FormGroupLayout>
+                field.onChange(finalValue);
+              }}
+              className={error ? "is-invalid" : ""}
+              inputProps={{ id }}
+              isLoading={isLoading}
+              options={options}
+              filterBy={() => true}
+              onSearch={(query) => {
+                void (async () => {
+                  setIsLoading(true);
+                  const results = await props.queryFn(query);
+                  setOptions(results);
+                  setIsLoading(false);
+                })();
+              }}
+              disabled={disabled}
+              onFocus={focusHandler}
+              {...reactBootstrapTypeaheadProps}
+            />
+          </FormGroupLayout>
+        )}
+      />
+      {icon && (
+        <FontAwesomeIcon
```

Comment:
May I suggest to use an `InputGroup` by reactstrap:

https://reactstrap.github.io/?path=/docs/components-inputgroup--input-group#multiple-addons


Probably best to create a new component called `InputGroupLayout` or add it to the existing `FormGroupLayout`.

Reply:
Done! Thank you, did not realize that all input uses the same layout!

Reply:
@ita-alibb 

Thanks for the changes. 

I would still suggest to use the `<InputGroup>` component from reactstrap because it looks quite a bit nicer. 

![image](https://github.com/neolution-ch/react-hook-form-components/assets/92933708/0529e7c0-fce3-48aa-be7c-b16c31d6dd3b)

One second thought, I think it also makes sense to keep this package light and flexible. Therfore I would suggest to allow any `ReactNode` to be supplied to two props in the `CommonInputProps` types:

```
addonLeft: ReactNode;
addonRight: ReactNode;
```

Than we can remove the font awesome dependency again from this package and leave it up to the consumer what he wants to supply (text, icons etc).